File: leetcode/1356. Sort Integers by The Number of 1 Bits/main.py
from typing import List


class Solution:
    def sortByBits(self, arr: List[int]) -> List[int]:
        return sorted(arr, key=lambda x: (self.countSetBits(x), x))

    def countSetBits(self, number: int) -> int:
        result = 0

        while number != 0:
            result += number & 1
            number >>= 1

        return result

File: leetcode/1356. Sort Integers by The Number of 1 Bits/main.ts
function sortByBits(arr: number[]): number[] {
  return arr.sort((a, b) => {
    let [bitsA, bitsB] = [countSetBits(a), countSetBits(b)];

    if (bitsA === bitsB) {
      return a - b;
    }

    return bitsA - bitsB;
  });
}

function countSetBits(num: number): number {
  let result = 0;

  while (num) {
    result += num & 1;
    num >>= 1;
  }

  return result;
}

console.log(sortByBits([0, 1, 2, 3, 4, 5, 6, 7, 8]))
